tests/Microsoft.Identity.Test.Unit/PublicApiTests/OBOTests.cs

```diff
@@ -347,6 +347,103 @@ public async Task AcquireTokenByObo_InitiateLongRunningProcessInWebApi_CacheKeyA
             }
         }
 
+        [TestMethod]
+        public async Task InitiateLongRunningObo_WithIgnoreCachedAssertion_TestAsync()
+        {
+            using (var httpManager = new MockHttpManager())
+            {
+                httpManager.AddInstanceDiscoveryMockHandler();
+
+                AddMockHandlerAadSuccess(httpManager);
+
+                var cca = BuildCCA(httpManager);
+
+                var userCacheAccess = cca.UserTokenCache.RecordAccess();
+
+                string oboCacheKey = null;
+
+                // Token's not in the cache, searched by user assertion hash, retrieved from IdP, saved with the provided OBO cache key
+                var result = await cca.InitiateLongRunningProcessInWebApi(TestConstants.s_scope, TestConstants.DefaultAccessToken, ref oboCacheKey)
+                    .ExecuteAsync().ConfigureAwait(false);
+
+                Assert.AreEqual(TokenSource.IdentityProvider, result.AuthenticationResultMetadata.TokenSource);
+                Assert.AreEqual(CacheRefreshReason.NotApplicable, result.AuthenticationResultMetadata.CacheRefreshReason);
+                userCacheAccess.AssertAccessCounts(0, 1);
+
+                // Initiate with different user assertion and IgnoreCachedAssertion flag
+                // MSAL will not match on the assertion and return cached token
```

Comment:
```suggestion
                // MSAL will match on the assertion and return cached token
```
Shouldn't the assertion match and return cached token?

Reply:
Ignore this comment.